File: frontend/src/types/environment.ts
// ============================================================
// 环境管理类型
// ============================================================

export interface Environment {
  id: number
  project_id: number
  name: string
  services: ServiceEntry[]
  variables: VariableEntry[]
  headers: HeaderEntry[]
  created_at: string
  updated_at: string
}

export interface ServiceEntry {
  url?: string
  name?: string
}

export interface VariableEntry {
  key: string
  value: string
  enabled?: boolean
}

export interface HeaderEntry {
  key: string
  value: string
  enabled?: boolean
}

File: backend/app/models/environment.py
from datetime import datetime
from sqlalchemy import String, JSON, Integer, Boolean, DateTime, ForeignKey, func, UniqueConstraint
from sqlalchemy.orm import Mapped, mapped_column
from app.models.base import Base


class Environment(Base):
    __tablename__ = "environments"
    __table_args__ = (UniqueConstraint('project_id', 'name', name='uq_env_project_name'),)

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    project_id: Mapped[int] = mapped_column(Integer, ForeignKey("projects.id", ondelete="CASCADE"), nullable=False, index=True)
    name: Mapped[str] = mapped_column(String(100), nullable=False)
    base_url: Mapped[str] = mapped_column(String(500), nullable=True, default=None)
    auth_config: Mapped[dict] = mapped_column(JSON, nullable=True, default=None)
    services: Mapped[list] = mapped_column(JSON, default=list)
    variables: Mapped[list] = mapped_column(JSON, default=list)
    headers: Mapped[list] = mapped_column(JSON, default=list)

    is_default: Mapped[bool] = mapped_column(Boolean, default=False)
    sort_order: Mapped[int] = mapped_column(Integer, default=0)
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now(), onupdate=func.now())
